libcxx/include/__config

```diff
@@ -1490,6 +1490,16 @@ __sanitizer_verify_double_ended_contiguous_container(const void*, const void*, c
 #    define _LIBCPP_DISABLE_UBSAN_UNSIGNED_INTEGER_CHECK
 #  endif
 
+// ASan related macros
+// _LIBCPP_IF_ASAN is used to removed ASan related code, whenever compiled without ASan.
```

Comment:
Since this is not suuuuper clean, I would suggest keeping this inside `<string>` and name it something string-specific like `_LIBCPP_STRING_IF_ASAN`. Hopefully in the future we can just clean it up.

Reply:
Sorry, I suggested it be moved. But lets go with Louis suggestion.

Reply:
I don't have an opinion about file, but so long name makes it less readable and I'm not a fan of it. But if we keep it inside string file, I understand that may be the best case, so I just did that, just pointing it out.